framework/include/actions/ActionFactory.h

```diff
@@ -85,7 +85,7 @@ class ActionFactory
 
   std::string getTaskName(const std::string & action);
 
-  MooseSharedPointer<Action> create(const std::string & action, const std::string & parsing_syntax, InputParameters parameters);
+  MooseSharedPointer<Action> create(const std::string & action, const std::string & action_name, InputParameters parameters);
```

Comment:
"diff" is just the name given by the parser, it could just as easily create some random set of characters to use as the name.

See my comment below about "name" and "parser_syntax"